openedx/tests/xblock_integration/test_recommender.py

```diff
@@ -650,7 +651,7 @@ def attempt_upload_file_and_verify_result(self, test_case, event_name, content=N
         happens or is rejected as expected.
         """
         if 'magic_number' in test_case:
-            f_handler = StringIO(test_case['magic_number'].decode('hex'))
+            f_handler = io.BytesIO(bytes.fromhex(test_case['magic_number']))
```

Comment:
This will cause f_handler to be a BytesIO sometimes and string IO other times.  I think we should have it be a consistent type. 

Also, it looks like this PR is a duplicate of https://github.com/edx/edx-platform/pull/21809/files  I think just close this for now and I'll work to get the other one landed.  You can close the corresponding ticket too.